app/pulp/app/models/task.py

```diff
@@ -205,6 +206,22 @@ def set_failed(self, exc, einfo):
         self.result = exception_to_dict(exc, einfo.traceback)
         self.save()
 
+    def cancel(self):
+        """
+        Set the task to a canceled state if it is in an incomplete state.
+
+        This updates the :attr: `finished_at` attribute, sets the :attr: `state` to
+        :attr: `CANCELED`
+
+        """
+        if self.state not in TASK_INCOMPLETE_STATES:
```

Comment:
Isn't this the opposite of what we want? I assume incomplete states are "running" and "waiting". If the current state is one of those, then we can change it to canceled and save.